src/components/Header/Header.tsx

```diff
@@ -0,0 +1,105 @@
+import { type FC, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { ShoppingCartOutlined } from "@ant-design/icons";
+import { Badge, Button, Row, Col, Space, message, Typography, Input, Dropdown, Modal, Layout } from "antd";
+import { getCartGoods, getGoods, getIsAuthValue, getUserLogin } from "src/store";
+import { useSelector } from "react-redux";
+import { userActions, cartActions, goodActions } from "src/store";
+import { useAppDispatch } from "src/hooks/useAppDispatch";
+import type { Good } from "src/types/general";
+import { debounce } from "lodash";
+import css from "./header.module.css";
+
+export const Header: FC = () => {
+    const cartGoods = useSelector(getCartGoods);
+    const isAuth = useSelector(getIsAuthValue);
+    const dispatch = useAppDispatch();
+    const login = useSelector(getUserLogin);
+    const [searchedGoods, setSearchedGoods] = useState<Good[]>([]);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const goods = useSelector(getGoods);
+
+    useEffect(() => {
+        if (!isAuth && login) {
+            dispatch(cartActions.setCart());
+        }
+    }, [isAuth]);
+
+    useEffect(() => {
+        setSearchedGoods(goods);
+    }, [goods]);
+
+    const items = searchedGoods.length > 0 ? searchedGoods.map((good) => ({
+        key: good.id,
+        label: <Link to={`/product/${good.id}`}>{good.label}</Link>
+    })) : [{ key: "noResults", label: "Ничего не найдено, попробуйте изменить запрос" }]
+
+    const searchHandle = debounce((value: string) => {
+        if (value.trim().length > 0) {
+            dispatch(goodActions.serverRequest({ text: value }))
+        } else setSearchedGoods([])
+    }, 1_500)
+
+    const showModal = () => {
+        setIsModalOpen(true);
+    };
+
+    const handleOk = () => {
```

Comment:
handleOk не самое удачное название, т.к. приходится искать куда оно навешивается, чтобы понять что это за ок такой. Лучше называть по сути выполняемой логики
```suggestion
    const handleLogout = () => {
```